src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -79,7 +80,48 @@ public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLo
             return Run(commandName, host, telemetryLogger, onFirstRun, args, null);
         }
 
+        static Mutex _entryMutex = new Mutex(false, "{5CB26FD1-32DB-4F4C-B3DC-49CFD61633D2}");
         public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, Action<IEngineEnvironmentSettings, IInstaller> onFirstRun, string[] args, string hivePath)
+        {
+            bool holdingMutex = false;
```

Comment:
Seems like rather than adding this flag it could be:

if (!_entryMutex.WaitOne())
{
    //exit with failure
}

try
{
    //existing stuff
}
finally
{
    _entryMutex.ReleaseMutex();
}

Reply:
Yes, I'll make that change. I had added that flag when tracking down some abandoned mutex issues while testing and left if there for safety, but it's overkill.